extensions/eclipse-che-theia-testing-service/src/testing-service.d.ts

```diff
@@ -0,0 +1,97 @@
+        export function references(pluginID: string, resource: UriComponents, position: Position, context: ReferenceContext, token: CancellationToken): Promise<Location[] | undefined>;
+        export function signatureHelp(
+            pluginID: string, resource: UriComponents, position: Position, context: SignatureHelpContext, token: CancellationToken
+        ): Promise<SignatureHelp | undefined>;
+        export function hover(pluginID: string, resource: UriComponents, position: Position, token: CancellationToken): Promise<Hover | undefined>;
+        export function documentHighlights(pluginID: string, resource: UriComponents, position: Position, token: CancellationToken): Promise<DocumentHighlight[] | undefined>;
+        export function documentFormattingEdits(pluginID: string, resource: UriComponents,
+            options: FormattingOptions, token: CancellationToken): Promise<TextEdit[] | undefined>;
+        export function documentRangeFormattingEdits(pluginID: string, resource: UriComponents, range: Range,
+            options: FormattingOptions, token: CancellationToken): Promise<TextEdit[] | undefined>;
+        export function onTypeFormattingEdits(
+            pluginID: string,
+            resource: UriComponents,
+            position: Position,
+            ch: string,
+            options: FormattingOptions,
+            token: CancellationToken
+        ): Promise<TextEdit[] | undefined>;
+        export function documentLinks(pluginID: string, resource: UriComponents, token: CancellationToken): Promise<DocumentLink[] | undefined>;
+        export function codeLenses(pluginID: string, resource: UriComponents, token: CancellationToken): Promise<CodeLensSymbol[] | undefined>;
+        export function codeActions(
+            pluginID: string,
+            resource: UriComponents,
+            rangeOrSelection: Range | Selection,
+            context: CodeActionContext,
+            token: CancellationToken
+        ): Promise<CodeAction[] | undefined>;
+        export function documentSymbols(pluginID: string, resource: UriComponents, token: CancellationToken): Promise<DocumentSymbol[] | undefined>;
+        export function workspaceSymbols(pluginID: string, query: string, token: CancellationToken): PromiseLike<SymbolInformation[]>;
+        export function foldingRange(
+            pluginID: string,
+            resource: UriComponents,
+            context: FoldingContext,
+            token: CancellationToken
+        ): PromiseLike<FoldingRange[] | undefined>;
+        export function documentColors(pluginID: string, resource: UriComponents, token: CancellationToken): PromiseLike<RawColorInfo[]>;
+        export function renameEdits(pluginID: string, resource: UriComponents, position: Position, newName: string, token: CancellationToken): PromiseLike<WorkspaceEditDto | undefined>;
+    }
+
+}
```

Comment:
Hi, are you following some rules to have sometimes `Promise` vs `PromiseLike` ?

Reply:
I this case, it should reflect what ever the "non-testing" version of the function uses.

Reply:
ok b/c I see in "non-testing"

```typescript
provideRenameEdits(document: TextDocument, position: Position, newName: string, token: CancellationToken): ProviderResult<WorkspaceEdit>

provideDocumentSymbols(document: TextDocument, token: CancellationToken): ProviderResult<SymbolInformation[] | DocumentSymbol[]>;
```

and `renameEdits` is using PromiseLike while `documentSymbols` is using Promise

Reply:
Types are different because when I was creating the d.ts file I used the types that came out of https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/common/plugin-api-rpc.ts#L1146 not the types that came out of the theia.d.ts

Reply:
yes I see that but we should map to external/public stuff, not implementation, no ?